### DISCUSSION TITLE
总算找到问题出在哪了

### DISCUSSION THREAD

**Discussion by BTMuli** [Arondight/Adachi-BOT#965] [Q&A] [answered]
问问你们的 webstorm 编码配置&powershell配置 & tabby 配置是怎样的，我这边跑程序，读 `process.argv`出来个 `娴嬭瘯`
![image](https://user-images.githubusercontent.com/72692909/203606648-e75dda82-387e-47e0-8cd2-3e67e3fc2b3c.png)
然后就发现我输入法打出来的`测试`是 GBK 编码然后 process 读的 UTF8 ...

**Comment by BTMuli** [comment]
尝试将系统编码转成UTF8之后我输入中文读不出`process.argv`了...
目前姑且找到一个折中的做法：
![image](https://user-images.githubusercontent.com/72692909/203612202-7f1de2ac-5e1f-4a55-ba57-577749d68f3c.png)
![image](https://user-images.githubusercontent.com/72692909/203612290-5c6c8c43-c0ca-4d05-8a26-fdd8a379fb3e.png)

**Reply by Arondight** [reply]
你可以看看能不能把 ws 的终端程序改成自己安装的 git bash

**Reply by BTMuli** [reply]
我试过Tabby跑Git bash也是不行，实在不行我用wsl克隆一下仓库进行开发算了，~~虽然ws读不了wsl的GPG~~

**Reply by mark9804** [reply]
Windows 我记得自己有一个挺好用的终端来着，tabby 有什么特别好用的功能吗

**Reply by BTMuli** [reply]
两者其实差不多，但是terminal之前我折腾过导致升win11后打开自动报错崩溃了，~~还有就是tabby对ssh支持挺强的但是我用不上~~

**Reply by Arondight** [reply]
我试了下我的 ws 里也运行正常啊

**Reply by mark9804** [reply]
会不会是输入法的问题（无端联想）

**Reply by Arondight** [reply]
哦，我没用 tabby ，ws 下我用的 cmd.exe ，外面我右键选在在 git bash 中打开的目录

**Reply by BTMuli** [reply]
xs，之前有段时间我也能跑，那段时间我提了3.1的pr跟文档补充

**Reply by BTMuli** [reply]
> 会不会是输入法的问题（无端联想）

搜狗输入法跟微软默认输入法应该没什么问题吧（）

> ~我们项目里只有你是Windows~
> 
> 可以临时切换代码页，当年用cmd的时候是
> 
> ```
> cp xxxx
> ```
> 
> 可以搜一下 powershell change code page

**Reply by mark9804** [reply]
其实我觉得大概率是tabby的问题

**Reply by BTMuli** [reply]
> 其实我觉得大概率是tabby的问题

下午试试把tabby卸了，换成win11的terminal看看，顺便也试试wsl

**Comment by mark9804** [comment]
~~我们项目里只有你是Windows~~

可以临时切换代码页，当年用cmd的时候是
```
cp xxxx
```

可以搜一下 powershell change code page

**Reply by Arondight** [reply]
我也在用 windows ，但是我在 git bash 里跑的命令行程序

**Comment by BTMuli** [comment, marked as answer]
问群友得出来的结果![2b4e76f2afff2e99_edit_1357211733990297.png](https://user-images.githubusercontent.com/72692909/203699927-acc381ce-63dd-49e9-b333-d07554bdf0ca.png)
我等会吃完饭回去试试

**Reply by BTMuli** [reply]
放弃了，直接跑node不调script了

发现一个调用米游社id生成用户卡片的site

![](https://genshin-card.getloli.com/rand/249066520.png)
